crates/extensions/c8y_mapper_ext/src/actor.rs

```diff
@@ -195,21 +200,9 @@ impl C8yMapperActor {
         mut message: EntityRegistrationMessage,
     ) -> Result<(), RuntimeError> {
         self.converter.append_id_if_not_given(&mut message);
-        // Convert and publish the registration message
-        let reg_messages = self.converter.convert_entity_registration_message(&message);
-        self.publish_messages(reg_messages).await?;
+        let reg_message = message.to_mqtt_message(&self.converter.mqtt_schema);
 
-        // Send the registration message to all subscribed handlers
-        self.publish_message_to_subscribed_handles(
-            &Channel::EntityMetadata,
-            message
-                .clone()
-                .to_mqtt_message(&self.converter.mqtt_schema)
-                .clone(),
-        )
-        .await?;
-
-        Ok(())
+        self.process_message(reg_message).await
```

Comment:
Since `process_registration_message()` is just 3 lines and called by only `process_registered_entities()`, I think it would be simpler if you remove `process_registration_message()` and move the contents to `process_registered_entites()`.

Reply:
Resolved by 6e76f88